cns/restserver/ipam.go

```diff
@@ -40,11 +39,9 @@ func (service *HTTPRestService) requestIPConfigHandlerHelper(ctx context.Context
 			},
 		}, errors.New("failed to validate ip config request")
 	}
-
 	// record a pod requesting an IP
 	service.podsPendingIPAssignment.Push(podInfo.Key())
-
-	podIPInfo, err := requestIPConfigsHelper(service, ipconfigsRequest)
+	podIPInfo, err := requestIPConfigsHelper(service, *ipconfigsRequest)
```

Comment:
@nddq I'm not sure why we need to pass context here according to linter error? it's not used in the functions mentioned

Reply:
ideally, we should pass the context all the way down but right now it is not necessary within this PR's scope so you can just add `// nolint:contextcheck` next to it